src/live.asp.net/Hubs/ChatHub.cs

```diff
@@ -0,0 +1,152 @@
+            <div class=""OnAirTextArea"" id=""chatWindow""></div>
+            <input type=""text"" id=""new-message""/></div></div>";
+        private const string QuestionDev = @"
+            <div class=""panel panel-primary OnAir"">
+            <div class=""panel-heading SmallPaddingNoMargin"">Ask a question</div>
+            <div class=""panel-body SmallPaddingNoMargin"">
+            <textarea id = ""question"" class=""OnAirTextArea""></textarea>
+            <button class=""btn btn-primary"" id=""ask"">Ask!</button>
+            </div></div>";
+        private readonly IOnAir _onAir;
+
+        public ChatHub(IOnAir onAir)
+        {
+            _onAir = onAir;
+        }
+
+        public void Send(string message)
+        {
+            foreach (Match match in Regex.Matches(message, "url:(.*?) "))
+            {
+                var url = match.Value.Replace("url:", "").Replace(" ", "");
+                message = message.Replace(match.Value, $"<a href=\"{url}\">{url}</a> ");
+            }
+            if (Context.User.Identity.IsAuthenticated)
+                Clients.All.UpdateMessages($"<b>{Context.User.Identity.Name}</b> {message}");
+            else
+            {
+                Clients.All.UpdateMessages(_onAir.GetUser(Context.ConnectionId).Username + " - " + message);
+            }
+        }
+
+        public void SetChatHandle(string chatHandle)
+        {
+            if (_onAir.UserExists(Context.ConnectionId))
+                _onAir.DelUser(Context.ConnectionId);
+            var user = new OnAirChatUsers {ConnectionId = Context.ConnectionId, IsAuth = false, Username = chatHandle};
+            _onAir.AddUser(user);
+
+            Clients.Caller.Join(
+                $"<div class=\"row\"><div class=\"row\"><div class=\"col-md-9\">{ChatDiv}</div ><div class=\"col-md-3\">{QuestionDev}</div ></div ></div >");
+        }
+
+        public void AskQuestion(string message)
+        {
+            var question = new OnAirQuestions
+            {
+                UserName =
+                    Context.User.Identity.IsAuthenticated
+                        ? Context.User.Identity.Name
+                        : _onAir.GetUser(Context.ConnectionId).Username,
+                Question = message
+            };
+
+            question = _onAir.AddQuestion(question);
+
+            var adminMenu =
+                $"<a href=\"{question.Id}\" id=\"{question.Id}\" action=\"Answer\" class=\"adminMenu\">Answer</a>";
+            //Working around "group issue with SignalR?"
+            foreach (var user in _onAir.GetUsers())
+            {
+                Clients.Client(user.ConnectionId).NewQuestion(question, user.IsAuth ? adminMenu : "");
+            }
+        }
+
+        public void UpdateQuestion(int id, string action)
+        {
+            var question = _onAir.GetQuestions().FirstOrDefault(q => q.Id == id);
+            if (question == null) throw new ArgumentNullException(nameof(question));
```

Comment:
What do you think about using .First(q => q.Id == id), or even it might be Single method?